src/dotnet/MLSearch.Service/Bot/Tools/ConversationToolsController.cs

```diff
@@ -0,0 +1,93 @@
+using System.Text;
+using ActualChat.Hashing;
+using ActualChat.Security;
+using ActualChat.Users;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using ActualChat.MLSearch.Engine;
+using ActualChat.MLSearch.Documents;
+using ActualChat.Chat;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Cors;
+using ActualChat.MLSearch.Bot.Tools.Context;
+using Microsoft.IdentityModel.Tokens;
+
+namespace ActualChat.MLSearch.Bot.Tools;
+
+[BotTools]
+[ApiController]
+[Route("api/bot/conversation")]
+[Produces("application/json")]
+public sealed class ConversationToolsController(ICommander commander, IBotToolsContextHandler botToolsContext, UrlMapper urlMapper): ControllerBase
+{
+    public sealed class Reply {
+        public required string Text {get; init;}
+    }
+
+    public sealed class ForwardLocalLinks {
+        public string? Comment {get; set;}
+        public required List<string> Links { get; init; }
```

Comment:
3. Inconsistent spaces in auto property declaration. (`{get; set;}` vs `{ get; init; }`)